Fails to compile with minimal version

During [minimal version testing](https://github.com/dekellum/body-image/pull/6) with a project using latest tokio-reactor v0.1.10, which now depends on parking_lot 0.9.0, this combination of dependencies is resolved by cargo:

```
parking_lot v0.9.0
├── lock_api v0.3.1
│   └── scopeguard v1.0.0
└── parking_lot_core v0.6.0
    ├── cfg-if v0.1.2
    ├── libc v0.2.55
    └── smallvec v0.6.10
```

But this combination fails to compile on either rustc 1.32.0 or a recent nightly, with the following errors:

``` txt
error[E0432]: unresolved imports `self::imp`, `crate::thread_parker::ThreadParker`, `crate::thread_parker::ThreadParker`
  --> /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.6.0/src/thread_parker/mod.rs:89:15
   |
89 | pub use self::imp::{thread_yield, ThreadParker, UnparkHandle};
   |               ^^^ could not find `imp` in `self`
   | 
  ::: /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.6.0/src/parking_lot.rs:8:28
   |
8  | use crate::thread_parker::{ThreadParker, ThreadParkerT, UnparkHandleT};
   |                            ^^^^^^^^^^^^
   | 
  ::: /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.6.0/src/word_lock.rs:9:28
   |
9  | use crate::thread_parker::{ThreadParker, ThreadParkerT, UnparkHandleT};
   |                            ^^^^^^^^^^^^

error: cannot find macro `__cfg_if_items!` in this scope
  --> /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.6.0/src/thread_parker/mod.rs:54:1
   |
54 | / cfg_if! {
55 | |     if #[cfg(all(has_sized_atomics, any(target_os = "linux", target_os = "android")))] {
56 | |         #[path = "linux.rs"]
57 | |         mod imp;
...  |
86 | |     }
87 | | }
   | |_^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

A reproduction of this local to the parking_lot tree, requires this change:

``` diff
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["concurrency"]
 edition = "2018"
 
 [dependencies]
-parking_lot_core = { path = "core", version = "0.6" }
+parking_lot_core = { version = "0.6" }
 lock_api = { path = "lock_api", version = "0.3.1" }
```

Followed by the following from the root of the parking_lot tree:

``` bash
cargo -Z minimal-versions generate-lockfile # on a nightly
cargo test
```

The obvious and simple fix for _minimal versions_ is to increase the minimum parking_lot_core dependency (for the next parking_lot crate release, e.g. 0.9.1) to 0.6.2.   Unless there is other suggestions I will PR that.  

Note: Updating cfg_if to some later version like 0.1.5 also avoids the issues but the cfg_if dependency is owned by parking_lot_core.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,7 @@ categories = ["concurrency"]
 edition = "2018"
 
 [dependencies]
-parking_lot_core = { path = "core", version = "0.6" }
+parking_lot_core = { path = "core", version = "0.6.2" }
 lock_api = { path = "lock_api", version = "0.3.1" }
 
 [dev-dependencies]

